### ISSUE TITLE
sbt project: `set every version` does not replace anything

### ISSUE THREAD

**Issue by dansanduleac** [sbt/sbt#1430]
The sbt project itself sets a build-wide version in `project/Sbt.scala`.
If I `set version in ThisBuild := "test"`, that replaces it. Good.

However, if I `set every version := "test"`, that overrides nothing:

```
> set every version := "test"
[info] Defining */*:version
[info] The new value will be used by no settings or tasks.
[info] Reapplying settings...
[info] Set current project to root (in build file:/D:/work/ext/sbt/)
```

Is this expected behaviour?

**Comment by jsuereth** [comment]
no, I think we just ran into this again.

**Comment by cunei** [comment]
Unless I messed up the bisection, "set every" started working erratically after commit ~~a3d89dc8656d7ee5fb2656dae79b5154ebe9a5e2~~ 322f6de6551665cade7d56b532348ea5dc3d54db

**Comment by cunei** [comment]
The definition of setAll(), which implements 'set every' uses `._1s`, which just grabs the keySet from the relation's forwardMap. Hence, having a mapping from something to an empty set, or not having the mapping, is /not/ equivalent when using `_1s`. I am issuing a pull request to revert the optimization.

**Comment by dwijnand** [comment]
With #1591 merged (and released as a part of 0.13.7) this issue is resolved and can therefore be closed.
